manifests/repomd_gpg.pp

```diff
@@ -0,0 +1,60 @@
+# == Class: certs
+# Sets up the GPG key for Pulp repository metadata signing
+class certs::repomd_gpg (
+  $enable                  = $::certs::enable_repomd_gpg,
+  $regenerate              = $::certs::regenerate_repomd_gpg,
+  $dir                     = $::certs::repomd_gpg_dir,
+  $user                    = $::certs::repomd_gpg_user,
+  $group                   = $::certs::repomd_gpg_group,
+  $gpg_name                = $::certs::repomd_gpg_name,
+  $comment                 = $::certs::repomd_gpg_comment,
+  $email                   = $::certs::repomd_gpg_email,
+  $key_type                = $::certs::repomd_gpg_key_type,
+  $key_length              = $::certs::repomd_gpg_key_length,
+  $expire_date             = $::certs::repomd_gpg_expire_date,
+  $batch_params            = $::certs::repomd_gpg_batch_params,
+) {
+  if $enable {
+
+    $gpg_name_param = $gpg_name ? { undef => '', default => "Name-Real: ${gpg_name}\n", }
+    $gpg_comment_param = $comment ? { undef => '', default => "Name-Comment: ${comment}\n", }
+    $gpg_email_param = $email ? { undef => '', default => "Name-Email: ${email}\n", }
+    $gpg_batch_params = $batch_params ? { undef => '', default => "${batch_params}\n", }
+    $gpg_params = "Key-Type: ${key_type}\nKey-Length: ${key_length}\nExpire-Date: ${expire_date}\n${gpg_name_param}${gpg_comment_param}${gpg_email_param}${gpg_batch_params}%no-protection\n%commit\n"
+    if $regenerate {
+      exec { 'delete existing repomd gpg':
+        onlyif  => "/usr/bin/test -e '${dir}/private-keys-v1.d'",
+        command => "rm -rf '${dir}'",
+        before  => File[$dir],
+      }
+    }
+    file { $dir:
+      ensure => directory,
+      owner  => $user,
+      group  => $group,
+      mode   => '0700',
+    } ->
+    exec { 'generate repomd gpg':
+      unless   => "/usr/bin/test -e '${dir}/secring.gpg' -o -e '${dir}/private-keys-v1.d'",
+      user     => $user,
+      group    => $group,
+      provider => shell,
+      command  => "echo '${gpg_params}' | /usr/bin/gpg --homedir '${dir}' --gen-key --batch",
+    }
+
+    $pub_file = "${dir}/pub.gpg"
+    $fp_cmd_print = "/usr/bin/gpg --homedir '${dir}' --list-secret-keys --with-colons"
+    $fp_cmd_parse = "/usr/bin/awk -F: '\$1 == \"fpr\" {print \$10; exit;}'"
```

Comment:
Are we guaranteed awk and gpg exist on every system? Or do we need some package dependecies?